payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripePaymentProcessor.kt

```diff
@@ -223,63 +265,75 @@ class StripePaymentProcessor : PaymentProcessor {
                 ProfileInfo(customer.delete().id)
             }
 
-    /* The 'expand' part will cause an immediate attempt at charging for the
-       subscription when creating it. For interpreting the result see:
-       https://stripe.com/docs/billing/subscriptions/payment#signup-3b */
-    override fun createSubscription(planId: String, stripeCustomerId: String, trialEnd: Long, taxRegionId: String?): Either<PaymentError, SubscriptionDetailsInfo> =
+    override fun createSubscription(planId: String, stripeCustomerId: String, trialEnd: Long, taxRegionId: String?): Either<PaymentError, SubscriptionPaymentInfo> =
             either("Failed to subscribe customer $stripeCustomerId to plan $planId") {
                 val item = mapOf("plan" to planId)
                 val taxRates = getTaxRatesForTaxRegionId(taxRegionId)
                         .fold(
                                 { emptyList<TaxRateInfo>() },
                                 { it }
                         )
-                val subscriptionParams = mapOf(
+                val trialEndInEpochSeconds = ofEpochMilliToSecond(trialEnd)
+                val params = mapOf(
                         "customer" to stripeCustomerId,
                         "items" to mapOf("0" to item),
-                        *(if (trialEnd > Instant.now().epochSecond)
-                            arrayOf("trial_end" to trialEnd.toString())
+                        *(if (trialEndInEpochSeconds > Instant.now().epochSecond)
```

Comment:
What does the asterisk mean in this context? This is necessarily obscure.

Reply:
Kotlins [spread](https://kotlinlang.org/docs/reference/functions.html#variable-number-of-arguments-varargs) operator.